core/rawdb/chain_iterator.go

```diff
@@ -206,6 +202,11 @@ func iterateTransactions(db ethdb.Database, from uint64, to uint64, reverse bool
 // There is a passed channel, the whole procedure will be interrupted if any
 // signal received.
 func indexTransactions(db ethdb.Database, from uint64, to uint64, interrupt chan struct{}, hook func(uint64) bool, report bool) {
+	// Bor: If ancient data was pruned, adjust range accordingly.
```

Comment:
Maybe worth creating a function `AdjustRangeForBor()` and call that. Will be helpful during upstream